### ISSUE TITLE
Add another

### ISSUE THREAD

**Issue by whatterz** [ministryofjustice/mojdt-design-system-backlog#6]
**Service:** Claim for Crown court defence
**Phase:** Live
**Department:** Legal Aid Agency

## Add or remove multiple items to or from a list
Use when
- a user needs to add to a list of things
- the form fields are in a set, for example, defendant details, supplier details, travel expenses

This is an alternative pattern to adding items to a tabular list via a circular interaction.

### Add defandant details
**GIVEN** a provider (advocate or litigator)
**WHEN** I enter a claim for Crown court fees
**I NEED** to provide all defendant details related to the case
**AND** their representation order details

**Example:**
![cccd_interaction_add_defendants](https://user-images.githubusercontent.com/23444/49007883-706cc400-f164-11e8-8d6f-61f9811de5ac.gif)

### Add supplier details
**GIVEN** a provider (litigator)
**WHEN** I create my account
**I NEED** to enter all my firm's office locations
**SO THAT** they can be used in claims for Crown court fees

**Example:**
![cccd_interaction_firms_add_suppliers](https://user-images.githubusercontent.com/23444/49007919-87131b00-f164-11e8-8d21-2f6300bb60af.gif)

### Add miscellaneous fees (Advocates)
**GIVEN** a provider (advocate)
**WHEN** I enter a claim for Crown court fees
**I NEED** to be able to enter multiple fees

**Example:**
![cccd_interaction_advocates_miscellaneous_fees](https://user-images.githubusercontent.com/23444/49008044-ce011080-f164-11e8-8eae-d0f826ff16b3.gif)

### Add miscellaneous fees (Litigators)
**GIVEN** a provider (litigator)
**WHEN** I enter a claim for Crown court fees
**I NEED** to be able to enter multiple fees

**Example:**
![cccd_interaction_litigators_miscellaneous_fees](https://user-images.githubusercontent.com/23444/49008063-d6f1e200-f164-11e8-81c4-2b37d415fc91.gif)

**Comment by reyrrep** [comment]
Service: Moving People Safely
Phase: Live Beta
Department: MoJ
![add_and_remove_mps_offences](https://user-images.githubusercontent.com/24682124/49238205-d45aeb00-f3f7-11e8-8b6a-6bfffd307057.gif)

**Comment by AnnaBrennan** [comment]
As part of our work on the **Form Builder** project we researched and designed two forms (LAA IOJ and HMCTS EX107) which both required the "**Add another**" component. Off the back of this work i've drafted a design pattern for this which i'd like your feedback on. 

[Add another component (please review)](https://docs.google.com/document/d/1tjuA48OEyRqH9BpLFibE_O76e9kMXDpvcza87foA-AI/edit?usp=sharing)

_NOTE: As you'll see "add a page" is outstanding, i'm waiting to see the solution that has already been designed in C100 as this will inform this section._

**Comment by helena-moj** [comment]
@AnnaBrennan makes sense to me that the numbering of the items only appears once the you have chosen to add another. I also think the text 'Add another [thing]' works, I have used it in the past and it seems to make sense to people, as does 'Remove'. I would say that the 'Remove' links could do with being closer to the input they are related to, especially when you are adding individual fields (maybe a gap of about 30px).  I would keep the font size of the links all the same and use the standard link styling, visually they are just missing the underline.

**Comment by helena-moj** [comment]
**Service:** Book returns and movements
**Phase:** Beta
**Department:** Home Office

In this example when you click 'Add another person' you go through a series of pages rather than entering information on the same page. 

![design-remove-person](https://user-images.githubusercontent.com/43538843/49518051-93e6eb80-f895-11e8-9e31-a69137264c50.png)

**Comment by helena-moj** [comment]
**Service:** People finder
**Phase:** Live
**Department:** Ministry of Justice

This one is pretty hard to use as you might imagine, but it's another example for the mix!

![image](https://user-images.githubusercontent.com/43538843/49518940-bd087b80-f897-11e8-93f1-fbd7ed39ce0d.png)

**Comment by helena-moj** [comment]
**Service:** Book returns and movements
**Phase:** Beta
**Department:** Home Office

![rmp-connecting-flights](https://user-images.githubusercontent.com/43538843/49519455-eaa1f480-f898-11e8-8516-55bb8e6ab907.png)

**Comment by helena-moj** [comment]
Related item in GOV.UK Design Systems backlog https://github.com/alphagov/govuk-design-system-backlog/issues/21

**Comment by AnnaBrennan** [comment]
@helena-moj Thanks for your feedback and comments on the "Add another" drafted pattern. You're feedback on the placement of the "Remove" link iterates the feedback we got from GDS this week; the current placement of Remove would be problematic for users with accessibility issues. They mentioned the "view through a straw" test... imagining if a page is magnified to the max then "remove" would be missed. We will revise the placement of this link.

The "Add another person" (Add another page) example above is also helpful as we're about to add this feature into the pattern. Will let you know when its ready for further review. Thanks.

**Comment by adamsilver** [comment]
We've added the add another component to the DS.

**Comment by gavinwye** [comment]
In the Design System as [Add another](https://moj-design-system.herokuapp.com/components/add-another)

**Comment by chrisadesign** [comment]
This component requires JS to work, suggest updating the guidance to make this clear.

**Comment by adamsilver** [comment]
It can and should work without JS too as long as the server is setup to
handle certain requests.
